File: print_speed.m
function print_speed(i, t, n)
%% printf speed of for loop
%% i is index; t is used time; n is total number.
%% by Hongwei Qin on Dec. 3, 2015
average_time = t / i;
remaining_time = (n - i) * average_time;
remaining_hour = floor(remaining_time / 3600);
remaining_min = round(remaining_time / 60 - remaining_hour * 60);
fprintf('%d / %d \t speed: %.3fs/iter, %d:%d (H:M) to go\n', i, n, average_time, remaining_hour, remaining_min); 
end

File: python/print_speed.py
# -*- coding: utf-8 -*-
import logging
import math

def print_speed(i, t, n):
    """print_speed(index, time, total_iteration)"""
    average_time = t / i
    remaining_time = (n - i) * average_time
    remaining_day = math.floor(remaining_time / 86400)
    remaining_hour = math.floor(remaining_time / 3600 - remaining_day * 24)
    remaining_min = round(remaining_time / 60 - remaining_day * 1440 - remaining_hour * 60)
    logging.info('%d / %d, speed: %.3f s/iter, ETA %d:%02d:%02d (D:H:M)\n' % (i, n, average_time, remaining_day, remaining_hour, remaining_min))
